storage/src/tests/persistence/filestorage/filestormanagertest.cpp

```diff
@@ -218,6 +219,57 @@ struct TestFileStorComponents {
     }
 };
 
+struct FileStorHandlerComponents {
+    DummyStorageLink top;
+    DummyStorageLink* dummyManager;
+    ForwardingMessageSender messageSender;
+    documentapi::LoadTypeSet loadTypes;
+    FileStorMetrics metrics;
+    std::unique_ptr<FileStorHandler> filestorHandler;
+
+    FileStorHandlerComponents(FileStorManagerTest& test, uint32_t threadsPerDisk = 1)
+        : top(),
+          dummyManager(new DummyStorageLink),
+          messageSender(*dummyManager),
+          loadTypes("raw:"),
+          metrics(loadTypes.getMetricLoadTypes()),
+          filestorHandler()
+    {
+        top.push_back(std::unique_ptr<StorageLink>(dummyManager));
+        top.open();
+
+        metrics.initDiskMetrics(loadTypes.getMetricLoadTypes(), 1, threadsPerDisk);
+
+        filestorHandler = std::make_unique<FileStorHandlerImpl>(messageSender, metrics, test._node->getComponentRegister());
+        filestorHandler->setGetNextMessageTimeout(50ms);
+    }
+    ~FileStorHandlerComponents() {}
+};
+
+struct PersistenceHandlerComponents : public FileStorHandlerComponents {
+    ServiceLayerComponent component;
+    BucketOwnershipNotifier bucketOwnershipNotifier;
+    std::unique_ptr<PersistenceHandler> persistenceHandler;
+
+    PersistenceHandlerComponents(FileStorManagerTest& test)
+        : FileStorHandlerComponents(test),
+          component(test._node->getComponentRegister(), "test"),
+          bucketOwnershipNotifier(component, messageSender),
+          persistenceHandler()
+    {
+        vespa::config::content::StorFilestorConfig cfg;
+        persistenceHandler =
+                std::make_unique<PersistenceHandler>(test._node->executor(), component, cfg,
+                                                     test._node->getPersistenceProvider(),
+                                                     *filestorHandler, bucketOwnershipNotifier,
+                                                     *metrics.disk->threads[0]);
+    }
+    std::unique_ptr<DiskThread> make_disk_thread() {
+        return createThread(*persistenceHandler, *filestorHandler, component);
+    }
+    ~PersistenceHandlerComponents() {}
```

Comment:
Consider `= default`

Reply:
That was not enough in this case.

Reply:
Move `PersistenceHandlerComponents::~PersistenceHandlerComponents() = default;` outside the class itself and the inliner becomes happy again

Reply:
Yes, that is the correct approach. = default in class definition applies the same inliner heuristics as not defining the destructor. Defining and empty destructor with {} changes the heuristics, while doing outside the class definitions avoids the problem.

Reply:
Fixed.